### PR TITLE
feat(ai): add transcription support

### LINKED ISSUE #963: feat support where has with throgh in orm
### Before feedback (在反馈之前)

- [x] There are no features that I want to submit in Issues (当前 Issues 中没有我将要提交的新功能)

### Describe Feature (描述功能)

current there no support HasOneThrough / HasManyThrough
current With / Load / LoadMissing only support gorm preload./

should support with in HasOne / HasMany / BelongsTo / Many2Many / MorphOne / MorphMany / **HasOneThrough** / **HasManyThrough**
should support 
- Has
- OrHas
- DoesntHave
- OrDoesntHave
- WhereHas
- OrWhereHas
- WhereDoesntHave
- OrWhereDoesntHave
- HasMorph
- OrHasMorph
- DoesntHaveMorph
- OrDoesntHaveMorph
- WhereHasMorph
- OrWhereHasMorph
- WhereDoesntHaveMorph
- OrWhereDoesntHaveMorph
- WithAggregate
- WithCount
- WithMax
- WithMin
- WithSum
- WithAvg
- WithExists
- WithRelation
- WithoutRelation
- WithRelationOnly

### PR DESCRIPTION
**Pull request by hwbrzzl**
## Summary
- Add a fluent AI transcription API that lets applications choose a provider, override the model, set a language hint, apply a timeout, and request speaker diarization.
- Add transcription file helpers for uploads, paths, and storage-backed files, and return typed transcript text, segments, and usage metadata from the framework response.
- Implement OpenAI `/audio/transcriptions` support with default transcription models, diarized segment parsing, provider capability validation, and regression coverage for nil-file and empty-response cases.

Closes goravel/goravel#963

## Why
This branch adds the missing speech-to-text workflow to Goravel's AI package, so applications can transcribe audio through the same fluent request style already used for chat, audio, and image generation. The new contracts and helpers make it possible to pass uploaded files or stored audio directly into a provider-backed transcription request without dropping down to provider-specific code.

```go
audio, err := ctx.Request().File("audio")
if err != nil {
	return ctx.Response().Json(http.StatusBadRequest, http.Json{"message": err.Error()})
}

response, err := facades.AI().Transcription(
	transcription.FromUpload(audio),
).Provider("openai").
	Language("en").
	Diarize().
	Timeout(30 * time.Second).
	Generate()
if err != nil {
	return ctx.Response().Json(http.StatusInternalServerError, http.Json{"message": err.Error()})
}

return ctx.Response().Success().Json(http.Json{
	"text":     response.Text(),
	"segments": response.Segments(),
})
```

The OpenAI implementation now handles both standard and diarized transcription responses, preserves usage metadata, and rejects invalid file inputs before making a provider request. That keeps the new API aligned with the rest of the framework while covering the edge cases and follow-up fixes added during review.

### CODE REVIEW

File: ai/openai/provider.go [hunk 1]
```diff
@@ -341,6 +379,25 @@ func (r *Provider) resolveAudioModel(model string) string {
 	return r.config.Models.Audio.Default
 }
 
+func (r *Provider) resolveTranscriptionModel(model string, diarize bool) string {
+	if model != "" {
+		return model
+	}
```
[the remark this answers is not in the document]
Comment (hwbrzzl) [reply ai/openai/provider.go:388]: If so, DefaultDiarizedTranscriptionModel will not always be used.

File: ai/transcription/transcription.go
Comment (hwbrzzl) [inline ai/transcription/transcription.go:1]: This file should be similar with `ai/image/`, `github.com/goravel/framework/facades` should not be used, the `Of` function is unnecessary.

File: ai/response_transcription_test.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
+package ai
+
+import (
+	"testing"
+	"time"
+
+	contractsai "github.com/goravel/framework/contracts/ai"
+	"github.com/stretchr/testify/assert"
```
Comment (hwbrzzl) [inline ai/response_transcription_test.go]:
```suggestion
	"github.com/stretchr/testify/assert"

	contractsai "github.com/goravel/framework/contracts/ai"
```

Comment (hwbrzzl) [reply ai/response_transcription_test.go]: Reordered the imports in `ai/response_transcription_test.go` to match the project import grouping.

File: ai/transcription/transcription.go
Comment (hwbrzzl) [reply ai/transcription/transcription.go:1]: Updated `ai/transcription/transcription.go` to match the attachment helper pattern used by `ai/image/file.go` and `ai/document/file.go`: I removed the `facades` dependency, dropped `Of`, and made `FromPath`/`FromStorage`/`FromUpload` return attachments instead of requests. The helper tests were rewritten around attachment behavior.

File: ai/openai/provider.go [hunk 3]
```diff
@@ -799,6 +856,77 @@ func (r *Provider) parseAudioResponse(response *http.Response, format goopenai.A
...
+		return nil, errors.AITranscriptionResponseIsEmpty
+	}
+
+	segments, err := r.resolveTranscriptionSegments(response, diarize)
+	if err != nil {
+		return nil, err
+	}
+	if response.Text == "" && len(segments) == 0 {
+		return nil, errors.AITranscriptionResponseIsEmpty
+	}
```
[the remark this answers is not in the document]
Comment (hwbrzzl) [reply ai/openai/provider.go]: Updated `parseTranscriptionResponse` to use OpenAI field-presence metadata instead of treating `text == ""` as invalid, so valid empty transcripts now pass while truly missing text and segments still return `AITranscriptionResponseIsEmpty`. I also added a regression test for `{"text":""}` responses.

File: ai/setup/stubs.go [hunk 4]
```diff
@@ -26,15 +26,21 @@ func init() {
...
+						},
+						"audio": map[string]any{
+							"default": "",
+						},
+						"transcription": map[string]any{
+							"default": "",
+						},
+						"image": map[string]any{
+							"default": "",
 					},
```
[the remark this answers is not in the document]
Comment (hwbrzzl) [reply ai/setup/stubs.go]: Fixed the raw config stub indentation so newly generated `config/ai.go` files are aligned with the surrounding map keys and stay gofmt-formatted when written out.

File: ai/openai/provider.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (hwbrzzl) [reply ai/openai/provider.go:388]: Adjusted `resolveTranscriptionModel` so diarized requests still fall back to `DefaultDiarizedTranscriptionModel` for the zero-config case, but honor `models.transcription.default` when a custom default is configured. I added focused tests for both paths.

[the remark this answers is not in the document]
Comment (hwbrzzl) [reply ai/openai/provider.go:388]: Right. The follow-up change keeps `DefaultDiarizedTranscriptionModel` as the fallback only for the zero-config/default case, but intentionally lets a custom `models.transcription.default` override it so configured provider-specific diarized models still work.

File: ai/openai/provider.go [hunk 5]
```diff
@@ -341,6 +379,32 @@ func (r *Provider) resolveAudioModel(model string) string {
...
+
+	return r.config.Models.Transcription.Default
+}
+
+func (r *Provider) resolveTranscriptionResponseFormat(diarize bool) goopenai.AudioResponseFormat {
+	if diarize {
+		return goopenai.AudioResponseFormatDiarizedJSON
+	}
+
+	return goopenai.AudioResponseFormatJSON
```
[the remark this answers is not in the document]
Comment (hwbrzzl) [reply ai/openai/provider.go:408]: This is accurate for the current OpenAI implementation: with `json`, non-diarized requests only return transcript text, so `Segments()` is only populated for diarized responses today. I updated the code path only for the typed-nil bug in this pass; for segments, I’d prefer to keep the current behavior and follow up separately if we want to switch non-diarized OpenAI requests to `verbose_json` and document any model-format trade-offs.

File: ai/transcription/transcription.go [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+func FromPath(path string, options ...contractsai.AttachmentOption) contractsai.Attachment {
+	return frameworkai.DocumentFromPath(path, options...)
+}
+
+func FromStorage(path string, options ...contractsai.AttachmentOption) contractsai.Attachment {
+	return frameworkai.DocumentFromStorage(path, options...)
+}
+
+func FromUpload(file contractsfilesystem.File, options ...contractsai.AttachmentOption) contractsai.Attachment {
+	return frameworkai.DocumentFromUpload(file, options...)
```
[the remark this answers is not in the document]
Comment (hwbrzzl) [reply ai/transcription/transcription.go:26]: Good catch. This was in the PR example rather than the implementation, and I updated the PR body to handle `ctx.Request().File("audio")` as `(filesystem.File, error)` before passing the file to `transcription.FromUpload`.

File: ai/openai/provider.go [hunk 7]
```diff
@@ -91,6 +97,38 @@ func (r *Provider) Audio(ctx context.Context, prompt contractsai.AudioPrompt) (c
 	return r.parseAudioResponse(response, params.ResponseFormat)
 }
 
+func (r *Provider) Transcription(ctx context.Context, prompt contractsai.TranscriptionPrompt) (contractsai.TranscriptionResponse, error) {
+	if prompt.File == nil {
+		return nil, errors.AITranscriptionFileRequired
+	}
+
+	content, err := prompt.File.Content(ctx)
```
[the remark this answers is not in the document]
Comment (hwbrzzl) [reply ai/openai/provider.go:108]: Updated `Provider.Transcription` to reject typed-nil `StorableFile` values with the same reflect-based nil check used elsewhere in `ai`, so we now return `AITranscriptionFileRequired` instead of risking a panic. I also added a regression test for a typed-nil attachment.

**Comment by hwbrzzl** [comment]
> ## Pull request overview
> Copilot reviewed 18 out of 22 changed files in this pull request and generated 2 comments.
> 
> Files not reviewed (4)

@copilot+claude-sonnet-4.6 review the rest files.

File: ai/image/image.go [hunk 8]
```diff
@@ -17,7 +16,3 @@ const (
 	SizePortrait  = contractsai.ImageSizePortrait
 	SizeLandscape = contractsai.ImageSizeLandscape
 )
```
[the remark this answers is not in the document]
Comment (hwbrzzl) [reply ai/image/image.go:18]: Ignore it

File: ai/transcription/transcription.go [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+func WithDisk(disk string) contractsai.AttachmentOption {
+	return frameworkai.WithDisk(disk)
+}
+
+func FromPath(path string, options ...contractsai.AttachmentOption) contractsai.Attachment {
+	return frameworkai.DocumentFromPath(path, options...)
+}
+
+func FromStorage(path string, options ...contractsai.AttachmentOption) contractsai.Attachment {
+	return frameworkai.DocumentFromStorage(path, options...)
```
[the remark this answers is not in the document]
Comment (hwbrzzl) [reply ai/transcription/transcription.go:22]: Ignore it
